pkg/cluster/ipaddresses.go

```diff
@@ -236,12 +236,22 @@ func (m *manager) ensureGatewayCreate(ctx context.Context) error {
 		return errors.New("private endpoint connection not found")
 	}
 
+	imageRegistryStorageAccountName, err := m.imageRegistryStorageAccountName()
+	if err != nil {
+		return err
+	}
+
+	err = validateImageRegistryStorageAccountName(imageRegistryStorageAccountName)
+	if err != nil {
+		return err
+	}
+
 	_, err = m.dbGateway.Create(ctx, &api.GatewayDocument{
 		ID: linkIdentifier,
 		Gateway: &api.Gateway{
 			ID:                              m.doc.OpenShiftCluster.ID,
 			StorageSuffix:                   m.doc.OpenShiftCluster.Properties.StorageSuffix,
-			ImageRegistryStorageAccountName: m.doc.OpenShiftCluster.Properties.ImageRegistryStorageAccountName,
+			ImageRegistryStorageAccountName: imageRegistryStorageAccountName,
```

Comment:
It is not directly related to the panic, but - does this mean that we probably have some clusters in prod where this field is empty in the  `GatewayDocument`?

I think we run `ensureGatewayUpgrade` & `ensureGatewayCreate` before `populateRegistryStorageAccountName`. So for older clustes I think we create 

cc @bennerv